### ISSUE TITLE
TestContainersUpdated events are not sent on project open/close

### ISSUE THREAD

**Issue by drognanar** [microsoft/nodejstools#2184]
##### Expected Behavior 
The TestContainerDiscoverer class is responsible for notifying the TestWindow of containers where tests should be discovered. It should call the TestContainersUpdated event whenever tests could potentially have changed.

My expectation is that OnProjectLoaded and OnProjectUnloaded should send the TestContainersUpdated event.

##### Actual Behavior
If you would put a breakpoint you can see that the OnTestContainersChanged is called, but the if condition evaluates to false and the event isn't actually sent, meaning that tests don't get removed on project close and don't get discovered on project open.

- NTVS Version: 1.5.10610.1 Commit Hash:529a87de2769e143655e31ea81795c42e5775ef8
- Visual Studio Version: 16.2 preview 4
- Node.js Version: 8.1.4

##### Steps to Reproduce
1. Open a nodejs project
2. Run build to discover tests
3. Unload the project
4. If the event would have been sent successfully the tests for the unloaded project should disappear

In the OnProjectUnloaded method the project is removed from `knownProjects` and this makes the if condition always fail, whereas in the OnProjectLoaded method the newly created project has HasRequestedContainers set to false making the if condition always fail.

**Comment by amcasey** [comment]
@armanio123 any chance your recent test discovery changes sorted this out?

**Comment by armanio123** [comment]
No. Vstest made some changes on the test explorer that impacted us. They have temporary fixed it but we still need to change how we discover tests and instead notify to the test window. I have already follow up with @drognanar offline.
